### DISCUSSION TITLE
Expose skippable frames via the API.

### DISCUSSION THREAD

**Discussion by rouzier** [klauspost/compress#670] [Ideas/Features]
Is it possible to add an API to use skippable frames.

https://github.com/facebook/zstd/blob/dev/doc/zstd_compression_format.md#skippable-frames

**Comment by klauspost** [comment]
@rouzier It is possible. You can of course prepend or append it to any encode.

Adding one is fairly simple. This is used for adding the final padding, and a standalone version could look like this:

```
// CreateSkippableFrame will add a skippable frame with the specified payload
// and append it to dst.
func CreateSkippableFrame(dst []byte, payload []byte) ([]byte, error) {
	if len(payload) == 0 {
		return dst, nil
	}
	if len(payload) > math.MaxUint32 {
		return dst, fmt.Errorf("requested skippable frame size (%d) > max uint32", len(payload))
	}
	dst = append(dst, 0x50, 0x2a, 0x4d, 0x18)
	f := uint32(len(payload))
	dst = append(dst, uint8(f), uint8(f>>8), uint8(f>>16), uint8(f>>24))
	dst = append(dst, payload...)
	return dst, nil
}
```
(untested, beware)

**Comment by klauspost** [comment]
Adding functionality to include it should also have callbacks for getting the payloads when decoding.
